akka-cluster-sharding/src/main/scala/akka/cluster/sharding/internal/EventSourcedRememberShards.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.internal
+
+import akka.actor.Props
+import akka.annotation.InternalApi
+import akka.cluster.sharding.ShardRegion.ShardId
+import akka.persistence.PersistentActor
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object EventSourcedRememberShards {
+  def props(typeName: String): Props =
+    Props(new EventSourcedRememberShards(typeName))
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] final class EventSourcedRememberShards(typeName: String) extends PersistentActor {
+
+  override val persistenceId: String = s"$typeName-remember-entitites"
+
+  private var shards = Set.empty[ShardId]
+
+  override def receiveRecover: Receive = {
+    case shardId: ShardId =>
+      // FIXME optimize for adding rather than reading (which is done only once)
+      shards += shardId
+  }
+
+  override def receiveCommand: Receive = {
+    case RememberEntitiesCoordinatorStore.GetShards =>
+      sender() ! RememberEntitiesCoordinatorStore.RememberedShards(shards)
+
+    case RememberEntitiesCoordinatorStore.AddShard(shardId: ShardId) =>
+      persistAsync(shardId) { shardId =>
+        shards += shardId
+        sender() ! RememberEntitiesCoordinatorStore.UpdateDone(shardId)
+      }
+  }
+
+}
```

Comment:
This does not cover how we'll migrate.

One idea I had was to make a persistent actor that uses the same id as a coordinator and accepts the events of that but only hangs on to the shard ids. Not sure it's a good idea thought.

Reply:
An alternative could be to have a migration version of the persistence based coordinator, that does the same, and then sends the full list of shards to the store when it has recovered. That way we wouldn't have to mix that logic into the regular event sourced store for the coordinator..

Reply:
As discussed let's merge this and i'll work on the migration